driver.go

```diff
@@ -126,14 +129,99 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 		maxap, err := mc.getSystemVar("max_allowed_packet")
 		if err != nil {
 			mc.Close()
-			return nil, err
+			return err
 		}
 		mc.maxAllowedPacket = stringToInt(maxap) - 1
 	}
 	if mc.maxAllowedPacket < maxPacketSize {
 		mc.maxWriteSize = mc.maxAllowedPacket
 	}
 
+	return err
+}
+
+//Connect opens a new connection without using a DSN
+func (c MySQLConnector) Connect(cxt context.Context) (driver.Conn, error) {
+	var err error
+
+	//Validate the connection parameters
+	//the following are required User,Pass,Net,Addr,DBName
+	//Pass may be blank
+	//The other optional parameters are not checks
+	//as GO will automatically enforce proper bool types on the options
+	if len(c.Cfg.User) > 32 || len(c.Cfg.User) <= 0 {
+		return nil, errInvalidUser
```

Comment:
Is this check really needed?
It seems ParseDSN() doesn't check user name length.  Empty user name is impossible?

Reply:
As far as I am aware an empty username will never authenticate. I guess it is not required press as you would just get a return from the server as a bad connection attempt however it does short cut a connection attempt thus using less resources/connections to the mysql server if the user forgets to provide a username. It also gives a very clear error message as to what is wrong.